microsoft/testsuites/performance/default_fio_testcases.json

```diff
@@ -0,0 +1,116 @@
+{
```

Comment:
We can use the logic of perf_premium_datadisks_4k as the default behavior instead of having this json file. Since perf_premium_datadisks_4k is already in use for Azure VM testing, I would say it is a very good default. This would also be in line with your point that the new test case is generic/configurable for version of perf_premium_datadisks_4k. In the absence of said configuration, it can fallback to perf_premium_datadisks_4k.

Reply:
That's fair. There are several things,

1. data driven test cases. If the parameters are only different things like this case. It's definitely should be data driven to share logics as much as possible. The shared data can be dict, or json, it doesn't matter.
2. For hard coded test cases names like `perf_premium_datadisks_4k`. It's necessary. When we test on different scenarios, we need standardized parameters to compare across multiple runs. It shouldn't need to ask every runner to choose from parameters.
3. For customizable parameters like this, it may be needed to handle occasional scenarios. But it shouldn't be overused to create incomparable results.

Reply:
To bring consistency in result, we are launching all test cases using perf_disk only now. If no param passed via runbook, we can launch some default params. This testcase is neither attached to specific scenario nor having specific requirements to run with.